tests/README.md

```diff
@@ -86,27 +49,79 @@ service (or your `QGIS_PGTEST_DB` connection string) uses a connection
 
-Instructions about writing tests for the processing framework
-can be found in a [separate README file](../python/plugins/processing/tests/README.md):
+    GH_WORKSPACE=$PWD \
+    docker-compose -f .docker/docker-compose-testing-postgres.yml up -d postgres
+    export PGHOST=`docker inspect docker_postgres_1 | jq -r .[0].NetworkSettings.Networks.docker_default.IPAddress`
+    export PGUSER=docker
+    export PGPASSWORD=docker
+    tests/testdata/provider/testdata_pg.sh
+    echo "${PGHOST} postgres # for qgis_test, docker" | sudo tee -a /etc/hosts
 
-    ${TOP_SRCDIR}/python/plugins/processing/tests/README.md
 
-Information about labeling tests design and organization:
+# Running the tests
 
-    ${TOP_SRCDIR}/tests/testdata/labeling/README.rst
+Once the test environment is setup, you can run all tests using `make check`.
+Note you will need `xvfb-run` for that (sudo apt-get install xvfb).
 
-WCS testing information can be found in:
+Individual tests can be run using `ctest`.
 
-    ${TOP_SRCDIR}/tests/testdata/raster/README.WCS
+For example if the output of `make check` ends like this:
 
-About benchmark tests you can read:
+```
+   The following tests FAILED:
+         77 - PyQgsLocalServer (Failed)
+```
 
-    ${TOP_SRCDIR}/tests/bench/README
+You could re-run the failing test with:
 
+```
+   ctest -V -R PyQgsLocalServer
+```
 
-# Run python tests in GDB
+The parameter `-V` enables verbose mode and `-R` takes a regular expression as
+parameter and will only run matching tests.
+
+Note that the name of the test as reported by `make check` does not
+always matches the name of the test as known by `cmake`. This is
+pretty annoying, but true. When this happens, the `ctest -R <guessedName>`
+command will report `No tests were found!!!` and you'll need to go
+hunting for the actual name. A useful hunting tool is `git grep`, which
+could be used like this:
+
+```
+  $ ctest -V -R testSslRequireNoCaCheck
+  No tests were found !!!
+  $ git grep testSslRequireNoCaCheck # Search for the reported testname
+  tests/src/python/test_authmanager_password_postgres.py:    def testSslRequireNoCaCheck(self):
+  $ git grep test_authmanager_password_postgres # Search for the file containing the reported testname
+  tests/src/python/CMakeLists.txt: ADD_PYTHON_TEST(PyQgsAuthManagerPasswordPostgresTest test_authmanager_password_postgres.py)
+  $ ctest -V -R PyQgsAuthManagerPasswordPostgresTest # use the CMakeLists.txt name
+```
+
+If you get `Could not connect to any X display` errors it means that your build
+machine does not have an X server.  In that case you need to run the test under
+`xvfb-run`.  For example:
+
+```
+    xvfb-run --server-args=-screen\ 0\ 1024x768x24 ctest -V -R PyQgsServerWMSGetMap
+```
+
+## Manually running python tests
+
+For python tests, you can run a specific test inside a unit file
+with something like this (from the build dir):
+
+```
+ export QGIS_PREFIX_PATH=${PWD}/output PYTHONPATH=${PWD}/output/python:$PYTHONPATH \
```

Comment:
if you already are in the build folder, I don't think the prefix path needs to be exported?

Reply:
Actually that line does not need export because it's passing the variable to the environment of the actual command.
See how you like 2fbd5a92153 - I don't know if the later section (using GDB) should be also rewritten to use ${builddir} and ${srcdir} consistently

Reply:
> Thank you for taking care of this readme

Thank you for reviewing it !